db/db_dynamic_level_test.cc

```diff
@@ -229,9 +229,17 @@ TEST_F(DBTestDynamicLevel, DynamicLevelMaxBytesBase2) {
     ASSERT_OK(Put(Key(static_cast<int>(rnd.Uniform(kMaxKey))),
                   RandomString(&rnd, 380)));
   }
+
+  rocksdb::SyncPoint::GetInstance()->LoadDependency({
+      {"CompactionJob::Run():Start", "DynamicLevelMaxBytesBase2:0"},
```

Comment:
The comment "Trigger a condition that the compaction changes base level and L0->Lbase happens at the same time." confuses me. Does this happen now?

Reply:
I think what it meant was that the following was supposed to happen:
1) compaction starts
2) flush starts
3) flush and compaction end at roughly the same time (or more realistically, flush finishes first and compaction finishes later)

It does this by writing enough data to fill L0, filling up the memtable again, enabling auto-compactions, and triggering a manual flush, in that order. I can put this description in the comment if that helps.

Reply:
Also, the base level does still change, since L4 fills up enough to push the base level up. The problem that sometimes happened was that L4 became too full if the compaction started after the manual flush, which pushed the base level up higher than it was expected to be.